Formula/synergy-core.rb

```diff
@@ -0,0 +1,376 @@
+    # This is not necessary on Linux. On Linux, the `cmake --install`
+    # command already places these binaries directly under `prefix/bin`.
+    if OS.mac?
+      bin.install_symlink prefix/"bundle/Synergy.app/Contents/MacOS/synergy"  # main GUI program
+      bin.install_symlink prefix/"bundle/Synergy.app/Contents/MacOS/synergys" # server
+      bin.install_symlink prefix/"bundle/Synergy.app/Contents/MacOS/synergyc" # client
+    end
+  end
+
+  service do
+    run [opt_bin/"synergy"]
+    run_type :immediate
+  end
+
+  def caveats
+    # The official Synergy binaries built by Symless (for which it charges money)
+    # are signed by a trusted code-signing certificate. For that reason, the user
+    # only has to add Synergy.app to the 'Accessibility' permissions list once,
+    # and it will work forever, even after upgrading or overwriting the app (so
+    # long as the new versions are also signed by the same certificate).
+    #
+    # Also, with the official binaries, the user only ever has to add one entry
+    # to the 'Accessibility' permissions list (namely, /Applications/Synergy.app),
+    # whereas with our version, it is sometimes necessary to add more than one
+    # program (or app bundle) to the permissions list (as explained below).
+    #
+    # This `caveats` block explains all of the side effects of our macOS binaries
+    # not being signed by a trusted certificate. This information is specific to
+    # the brew installation and is not included in upstream docs because upstream
+    # avoids many of these inconveniences by shipping binaries signed with a
+    # trusted certificate (and charging money for those binaries).
+    on_macos do
+      s = "On macOS, Synergy requires the 'Accessibility' permission.\n" \
+          "You can grant this permission by navigating to:\n" \
+          "  System Preferences -> Security & Privacy -> Privacy -> Accessibility\n" \
+          "Click the '+' button to grant the permission to new programs.\n\n" \
+          "If you run Synergy using `brew services`, grant the 'Accessibility' permission to:\n" \
+          "  #{opt_prefix}/bundle/Synergy.app\n\n"
+      if Hardware::CPU.arm?
+        s += "On Apple Silicon, you must also grant the 'Accessibility' permission to:\n" \
+             "  #{opt_prefix}/bundle/Synergy.app/Contents/MacOS/synergy\n\n"
+      end
+      s += "If you want to run Synergy in a terminal emulator, you must grant the 'Accessibility'\n" \
+           "permission to the terminal emulator."
+      if Dir.exist?("/System/Applications/Utilities/Terminal.app")
+        s += " For example, if you use Terminal.app, you must\ngrant the 'Accessibility' permission to:\n" \
+             "  /System/Applications/Utilities/Terminal.app"
+      end
+      s += "\n\n"
+      if Hardware::CPU.arm?
+        s += "On Apple Silicon, if you run Synergy under a shell (such as bash), you may also\n" \
+             "need to grant the 'Accessibility' permission to that shell.\n\n"
```

Comment:
If none of this is documented upstream you should really add a PR for that upstream because this is not specific to brew. It applies to everyone using the open-source version.

Reply:
Okay. In the meanwhile, I've reduced the caveats section to the bare minimum necessary to point users in the right direction. This is consistent with what some other things in brew have done (see, e.g., `brew info dnscrypt-proxy`, `brew info santa`). Arguably `dyscrypt-proxy` is more detailed by far than what I've written here.